idd/Energy+.idd.in

```diff
@@ -526,18 +540,18 @@ ShadowCalculation,
        \note only really applicable to RunPeriods
        \note warning issued if >31
   N2 , \field Maximum Figures in Shadow Overlap Calculations
-       \note Number of allowable figures in shadow overlap calculations
+       \note Number of allowable figures in shadow overlap in PolygonClipping calculations
        \type integer
        \minimum 200
        \default 15000
-  A2 , \field Polygon Clipping Algorithm
+  A3 , \field Polygon Clipping Algorithm
        \note Advanced Feature.  Internal default is SutherlandHodgman
        \note Refer to InputOutput Reference and Engineering Reference for more information
        \type choice
        \key ConvexWeilerAtherton
        \key SutherlandHodgman
        \default SutherlandHodgman
-  A3 , \field Sky Diffuse Modeling Algorithm
+  A4 , \field Sky Diffuse Modeling Algorithm
```

Comment:
Are there any parameters needed to control the pixel counting algorithm?  I would think that number of pixels would be a parameter?

Reply:
Would an option to write out images or renderings be useful for debugging? Ability to write out shading schedules? Any other reporting that should be considered?

Reply:
I think adding a resolution input would be good. The rendering is mapped to a square, so maybe:

```
\field Pixel Counting Resolution
\note Number of pixels in both dimensions of the surface rendering
\type integer
\default 512
```

The library has the ability to open a window rendering of the scene, but it is currently used only for debugging the C++ code (not for general simulations). This could be added later if needed.

Reply:
Does the pixel counting method cover diffuse shading?

Reply:
The method stretches any surface, regardless of its size, to fill as much of the rendered view as possible and provide the maximum possible resolution. What is rendered is often stretched in the X or Y direction to fit within the square of the determined size (that is, even a rectangular surface will be stretched to an aspect ratio of 1:1). The algorithm is aware that the projected dimensions are not necessarily equal and accounts for this appropriately.

Reply:
Yes, diffuse shading is done using the pixel counting method for each location in the sky dome grid.

Reply:
So - does this field have any impact when using the GPU method? Ultimately, when this is complete, IDD notes and I/O ref will need to highlight which fields apply only to certain shading methods.

Reply:
It will still have an impact. Ideally we will figure out how to handle variable transmittance surfaces and this will be handled the same way it is for the polygon clipping method.